r&d.md

```diff
@@ -1276,19 +1304,202 @@
   3. Each pre-eval call, :ref, :value
     - doesn't generally save the issue
 
-## [ ] Async effects init?
-
-  + Allows referring to newly created state from inside methods
-  + Errors in effects in first run don't break sprae
+  4. Outside of parent effect:
+    + effect can be async, so it must not cause async init
+
+## [x] Immediate state access #58 -> pre-create store
+
+  1. Async effects init
+    + Allows referring to newly created state from inside methods
+    + Errors in effects in first run don't break sprae
+      - we still brand errors
+    + Possible batch update
+    + Possible await in effects
+    - First blank frame
+    - delay in init
+    - all other frameworks are sync
+    - grok conclusion is no https://grok.com/share/bGVnYWN5_8d145672-9faf-4df9-8b5a-9f190ac7e58f
+      - size bloat, not sprae philosophy, existing alternatives
+
+  2. `x.async` prop modifier
+    - whole syntax conceptual space
+    + we already do `parts`
+    + it's `async func` directly
+
+  3. `sprae.async` flag
+    - unmade decision
+
+  4. `async.sprae` flavor
+    - not sufficient for a separate flavor
+    ~+ unless we make batch, async effects
+
+  5. Just export `sprae.store`
+
+## [x] Should we separate `k,v in b` to `k in b`, `v of b` -> no. Use v,k in b, as all other frameworks do
+
+  + likely perf optimizatino
+  + same as in JS
+  + we rarely need both key and value
+  + possibly less issue with store
+  + it's hard to remember if it's k,v or v,k
+  - contrary to all-frameworks convention
+  - js should've had it
+  - it's called `each`, not `for`
+
+### [x] ALT: keep only `k in b` -> no
+
+  + will simplify state management: k doesn't change, unlike item
+  + if you want separate scope - just create via `:with.item="items[i]"`
+    ? can we make `:let="item=items[i]"` instead of `:with`?
+
+## [ ] Conditional queries: md:onclick, lg:onclick etc?
```

Comment:
Nah, sounds bloaty, if someone really needs to do something different in an event based on screen size they should just handle it in the event handler function imho